specification/confidentialledger/Microsoft.CodeTransparency/models.tsp

```diff
@@ -0,0 +1,275 @@
+
+  ...Foundations.ErrorResponse;
+}
+
+#suppress "@azure-tools/typespec-azure-core/documentation-required"
+@doc("The Response body of the service parameters")
+model ParametersResult {
+  serviceCertificate: string;
+  serviceId: string;
+  signatureAlgorithm: string;
+  treeAlgorithm: string;
+}
+
+@doc("Response of entry submission containing the operationId.")
+model CreateEntryResult {
+  @doc("String representing the operation id submitted")
+  @pattern("^[0-9]+\\.[0-9]+$")
+  operationId: string;
+}
+
+@doc("An Entry in Cose format")
+model CoseEntry {
+  // defaults to json in the post method
+  @doc("The MIME content type a Cose body is application/cose, containing a CoseSign1 signature.")
+  @header
+  contentType: "application/cose";
+
+  @doc("A raw CoseSign1 signature")
+  @body
+  body: bytes;
+}
+
+@doc("A Raw Entry in cbor format")
+model RawEntry {
+  @doc("The MIME content type for raw receipt is application/cbor.")
+  @header
+  contentType: "application/cbor";
+
+  @doc("A raw CBOR content")
+  @body
+  body: bytes;
+}
+
+#suppress "@azure-tools/typespec-azure-core/casing-style"
+@doc("A List of Entry IDs")
+@pagedResult
+model EntryIds {
+  @doc("List of entry ids")
+  @items
+  transactionIds: string[];
+
+  @doc("Path to the next page of results")
+  @nextLink
+  nextLink?: string;
+}
+
+@doc("The EntryId parameter.")
+model EntryIdParameter {
+  @doc("ID of the entry to retrieve.")
+  @path
+  @pattern("^[0-9]+\\.[0-9]+$")
+  entryId: string;
+}
+
+@doc("The EmbedReceipt parameter.")
+model EmbedReceiptParameter {
+  @doc("Embed a receipt as part of the response")
+  @query
+  embedReceipt?: boolean = false;
+
+  ...EntryIdParameter;
+}
+
+@doc("Range of transactions for historical queries")
+model TransactionsRange {
+  @doc("Starting Transaction Id")
+  @query
+  @minValue(0)
+  from?: int64;
+
+  @doc("Ending Transaction Id")
+  @query
+  @minValue(0)
+  to?: int64;
+}
+
+@doc("The OperationId parameter.")
+model OperationIdParameter {
+  @doc("ID of the operation to retrieve.")
+  @pattern("^[0-9]+\\.[0-9]+$")
+  @path
+  operationId: string;
+}
+
+@doc("The Response body of the ListOperations API")
+model ListOperationResult {
+  @doc("List of operations")
+  operations: GetOperationResult[];
+}
+
+@doc("The Response body of the GetOperationId API.")
+model GetOperationResult {
+  @doc("ID of the transaction. Only if status is Succeeded.")
+  @pattern("^[0-9]+\\.[0-9]+$")
+  entryId?: string;
+
+  @doc("Error in json format. Only if Status is Failed.")
+  @lroErrorResult
+  error?: string;
+
+  @doc("OperationId")
+  @pattern("^[0-9]+\\.[0-9]+$")
+  operationId: string;
+
+  @doc("Status of the operation.")
+  status: OperationStatus;
+}
+
+#suppress "@azure-tools/typespec-azure-core/documentation-required" "https://github.com/Azure/typespec-azure/issues/3107"
+@doc("Possible statuses of the operation.")
+@lroStatus
+enum OperationStatus {
+  running,
+
+  @lroFailed
+  failed,
+
+  @lroSucceeded
+  succeeded,
+}
```

Comment:
Please use `union` for extensible enum.

Context https://github.com/Azure/azure-rest-api-specs/pull/27687